File: nodemanager/data/OutputData.cpp
#include "OutputData.h"

json::value OutputData::ToJson() const
{
    json::value j;
    j["NodeName"] = json::value::string(this->NodeName);
    j["Order"] = this->Order;
    j["Content"] = json::value::string(this->Content);
    j["Eof"] = this->Eof;

    return std::move(j);
}

File: src/NodeAgent/Models/OutputData.cs
using System.ComponentModel.DataAnnotations;

namespace NodeAgent.Models;

public class OutputData : DiagBase
{
    [Required]
    public string NodeName { get; set; } = default!;

    public int Order { get; set; } = 0;

    [Required]
    public string Content { get; set; } = default!;

    public bool Eof { get; set; } = false;
}
